web/pandas/pdeps/0013-standardize-apply.md

```diff
@@ -0,0 +1,158 @@
+
+Below is an overview of the current behavior in table form when giving callables to `agg`, `transform` and `apply`. As an example on how to read the tables, when a non-ufunc callable is given to `Series.agg`, `Series.agg` will first try to apply the callable to each element in the series, and if that fails, will fall back to call the series using the callable.
+
+(The description may not be 100 % accurate because of various special cases in the current implementation, but will give a good understanding of the current behavior).
+
+### agg
+
+|                                    | Series                           | DataFrame                        | groupby   |
+|:-----------------------------------|:---------------------------------|:---------------------------------|:----------|
+| ufunc or list/dict of ufuncs       | series                           | series                           | series    |
+| other callables (non ufunc)        | Try elements, fallback to series | series                           | series    |
+| list/dict of callables (non-ufunc) | Try elements, fallback to series | Try elements, fallback to series | series    |
+
+### transform
+
+|                                    | Series                           | DataFrame                        | groupby   |
+|:-----------------------------------|:---------------------------------|:---------------------------------|:----------|
+| ufunc or list/dict of ufuncs       | series                           | series                           | series    |
+| other callables (non ufunc)        | Try elements, fallback to series | series                           | series    |
+| list/dict of callables (non-ufunc) | Try elements, fallback to series | Try elements, fallback to series | series    |
+
+### apply
+
+|                                    | Series   | DataFrame   | groupby   |
+|:-----------------------------------|:---------|:------------|:----------|
+| ufunc or list/dict of ufuncs       | series   | series      | series    |
+| other callables (non ufunc)        | elements | series      | series    |
+| list/dict of callables (non-ufunc) | Try elements, fallback to series | series    | series    |
+
+The 3 tables show that:
+
+1. when given numpy ufuncs, callables given to `agg`/`transform`/`apply` operate on series data
+2. when used on groupby objects, callables given to `agg`/`transform`/`apply` operate on series data
+3. else, in some case it will try element-wise operation and fall back to series-wise operations if that fails, in some case will operate on series data and in some cases on element data.
```

Comment:
Sidenote: to be honest, this "in some cases on element data" is IMO by far the _main_ use case of `Series.apply`, and in this case (being passed a single function), it always operates element by element, consistently and unambiguously.